rgw/admin/radosgw.go

```diff
@@ -0,0 +1,157 @@
+package admin
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"errors"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+const (
+	authRegion                 = "default"
+	service                    = "s3"
+	connectionTimeout          = time.Second * 3
+	post              verbHTTP = "POST"
+	put               verbHTTP = "PUT"
+	get               verbHTTP = "GET"
+	delete            verbHTTP = "DELETE"
+)
+
+var (
+	errNoEndpoint  = errors.New("endpoint not set")
+	errNoAccessKey = errors.New("access key not set")
+	errNoSecretKey = errors.New("secret key not set")
+)
+
+type verbHTTP string
+
+// API struct for New Client
+type API struct {
+	AccessKey  string
+	SecretKey  string
+	Endpoint   string
+	HTTPClient *http.Client
+	Debug      bool
+}
+
+// New returns client for Ceph RGW
+func New(endpoint, accessKey, secretKey string, httpClient *http.Client) (*API, error) {
+	// validate endpoint
+	if endpoint == "" {
+		return nil, errNoEndpoint
+	}
+
+	// validate access key
+	if accessKey == "" {
+		return nil, errNoAccessKey
+	}
+
+	// validate secret key
+	if secretKey == "" {
+		return nil, errNoSecretKey
+	}
+
+	// set http client, if TLS is desired it will have to be passed with an http client
```

Comment:
:+1: I wouldn't even be against always requiring a valid http client be passed in, but this strikes me as a good compromise between flexibility and convenience.